flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -261,19 +263,33 @@ public void submitSQLJob(SQLJobSubmission job) throws IOException {
 		AutoClosableProcess.runBlocking(commands.toArray(new String[0]));
 	}
 
-	public void copyOptJarsToLib(String jarNamePrefix) throws FileNotFoundException, IOException {
-		final Optional<Path> reporterJarOptional;
-		try (Stream<Path> logFiles = Files.walk(opt)) {
-			reporterJarOptional = logFiles
-				.filter(path -> path.getFileName().toString().startsWith(jarNamePrefix))
+	public void moveJar(JarMove move) throws IOException {
```

Comment:
The proposed code is indeed over-engineered for current usages, but we will soon have operations from opt -> plugins and lib -> opt, and this approach can handle this easier since it doesn't require additional methods.

I will fix the method not doing a move.

Reply:
`moveJar` -> `copyJar`?